### PR TITLE
Always-not-synced exception in block syncer during import-db operation

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added an exception to the block syncer as to always consider itself not synchronized in the import-db process

Testing scenario: in the normal operation, the network should behave normally, the WARN message `using always-not-synced status because the node is running in import-db` should not appear on any node. During the import-db operation, the above WARN message should appear and the node should not wait more that 10 rounds between epochs on the metachain.
When testing the import-db operation, the branch `test-always-not-synced-in-import-db` should be used.

### CODE REVIEW

File: process/sync/metablock.go [hunk 1]
```diff
@@ -60,6 +60,11 @@ func NewMetaBootstrap(arguments ArgMetaBootstrapper) (*MetaBootstrap, error) {
 		uint64Converter:     arguments.Uint64Converter,
 		poolsHolder:         arguments.PoolsHolder,
 		indexer:             arguments.Indexer,
+		isInImportMode:      arguments.IsInImportMode,
+	}
+
+	if arguments.IsInImportMode {
```
Comment (SebastianMarian) [inline process/sync/metablock.go:66]: Maybe to be sure that isImportMode was correctly or was not missed to be set with the value of argument.IsImportMode you can check directly with isInImportMode in this condition

File: process/sync/shardblock.go [hunk 2]
```diff
@@ -57,6 +57,11 @@ func NewShardBootstrap(arguments ArgShardBootstrapper) (*ShardBootstrap, error)
 		uint64Converter:     arguments.Uint64Converter,
 		poolsHolder:         arguments.PoolsHolder,
 		indexer:             arguments.Indexer,
+		isInImportMode:      arguments.IsInImportMode,
+	}
+
+	if arguments.IsInImportMode {
```
Comment (SebastianMarian) [inline process/sync/shardblock.go:63]: The same here

File: process/sync/metablock.go (hunk 1 above)
Comment (iulianpascalau) [reply process/sync/metablock.go:66]: done

File: process/sync/shardblock.go (hunk 2 above)
Comment (iulianpascalau) [reply process/sync/shardblock.go:63]: done

**Review by SebastianMarian — APPROVED** [review]

File: process/sync/baseSync.go [hunk 3]
```diff
@@ -632,6 +632,15 @@ func (boot *baseBootstrap) syncBlock() error {
 	return nil
 }
 
+func (boot *baseBootstrap) shouldSync() bool {
+	if boot.isInImportMode {
+		return true
+	}
+
+	nodeState := boot.GetNodeState()
```
Comment (sasurobert) [inline process/sync/baseSync.go:640]: nice!

**Review by sasurobert — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]

**Pull request merged**